lib/pbench/agent/utils.py

```diff
@@ -188,3 +190,40 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+# From https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address
+_allowed = re.compile(
+    r"^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$"
+)
+
+
+def validate_hostname(host_name):
+    """validate_hostname - validate the given hostname uses the proper syntax.
+
+    A host name that follows RFC 952 (amended by RFC 1123) is accepted only.  Host
+    names are not resolved to IP addresses, and IP addresses are also accepted.
+
+    Algorithm taken from: https://stackoverflow.com/questions/2532053/validate-a-hostname-string
```

Comment:
This says there is a 63-octet limit per component: should you be checking for that?

Reply:
We can be a little loose with the hostname checking, this solves the most egregious issues.  We could validate 63 characters between dots, but it won't cause us any headaches.